### DISCUSSION TITLE
is there has a layer to encrypt/decrypt field before-write/after-read

### DISCUSSION THREAD

**Discussion by BrightVan** [simolus3/drift#2702] [Q&A]
we can implement encrypt/decrypt logic more Flexible

**Comment by simolus3** [comment]
If you have something that does the encryption in Dart, you could use a blob column with a [type converter](https://drift.simonbinder.eu/docs/advanced-features/type_converters/) doing the encryption / decryption steps.

**Reply by BrightVan** [reply]
encryption is time-consuming operation, but i can not return a future.
```dart
class CryptorConverter extends TypeConverter<String,String>{
  @override
  String fromSql(String fromDb)async {
     // can not return a future
    final s = await Cryptor.decode(fromDb);
    return s;
  }

  @override
  String toSql(String value) async{
    // can not return a future
    final s = await Cryptor.encode(value);
    return s;
  }
}
```
